src/udev/net/link-config.c

```diff
@@ -937,6 +938,61 @@ static int link_apply_sr_iov_config(Link *link, sd_netlink **rtnl) {
         return 0;
 }
 
+static int link_apply_rps_cpu_mask(Link *link) {
+        _cleanup_free_ char *mask_str, *dev_path, *rps_cpus_path = NULL;
+        _cleanup_closedir_ DIR *d = NULL;
+        LinkConfig *config;
+        int r;
+
+        assert(link);
+        config = ASSERT_PTR(link->config);
+
+        mask_str = cpu_set_to_mask_string(&config->rps_cpu_mask);
+        if (!mask_str)
+                return -ENOMEM;
+
+        log_link_debug(link, "Applying RPS CPU mask: %s", mask_str);
+
+        dev_path = path_join("/sys/class/net/", link->ifname, "queues");
+        if (!dev_path)
+                return log_oom();
+
+        d = opendir(dev_path);
+        if (!d)
+                return -errno;
+
+        /* Currently, this will set CPU mask to all rx queue of matched device. */
+        FOREACH_DIRENT(de, d, break) {
+                int queue;
+                const char *c;
+
+                if (de->d_type != DT_DIR)
+                        continue;
+
+                c = startswith(de->d_name, "rx-");
+                if (!c)
+                        continue;
+
+                r = safe_atoi(c, &queue);
```

Comment:
this is not going to be signed, is it? looks weird parsing this as a signed value then. use safe_atou()?